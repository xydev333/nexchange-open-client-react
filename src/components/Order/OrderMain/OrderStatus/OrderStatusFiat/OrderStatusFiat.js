import React from 'react';
import cx from 'classnames';
import Deposit from '../../images/deposit.svg';
import Confirmations from '../../images/confirmations.svg';
import Done from '../../images/done.svg';
import STATUS_CODES from 'StatusCodes';
import styles from '../OrderStatus.scss';

const OrderStatusFiat = props => {
  return (
    <div>
      <div
        className={
          [0, 8].indexOf(props.status) > -1
            ? styles.step
            : props.status > 11
              ? cx(styles.step, styles.done)
              : cx(styles.step, styles.active)
        }
        data-toggle="tooltip"
        data-placement="top"
        title=""
        data-original-title="In this step we are waiting for your deposit."
      >
        <Deposit />
        <h4>1. Awaiting deposit</h4>
      </div>

      <div
        className={
          STATUS_CODES[props.status] === 'PAID_UNCONFIRMED'
            ? cx(styles.step, styles.active)
            : props.status >= 13
              ? cx(styles.step, styles.done)
              : styles.step
        }
        data-toggle="tooltip"
        data-placement="top"
        title=""
        data-original-title="We have received your funds and are now waiting
                for proof of residence and government issued ID verification."
      >
        <Confirmations />
        <h4>2. Awaiting verification</h4>
      </div>

      <div
        className={
          props.status === 13 || props.status === 14
            ? cx(styles.step, styles.active)
            : props.status >= 15
              ? props.status === 15
                ? cx(styles.step, styles.active)
                : cx(styles.step, styles.done)
              : styles.step
        }
        data-toggle="tooltip"
        data-placement="top"
        title=""
        data-original-title="We got the funds and now have transferred our funds to you."
      >
        <Done />
        <h4>3. All done</h4>
      </div>
    </div>
  );
};

export default OrderStatusFiat;
