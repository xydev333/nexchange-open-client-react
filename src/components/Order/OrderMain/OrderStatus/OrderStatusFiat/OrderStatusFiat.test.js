import React from 'react';
import { shallow } from 'enzyme';
import OrderStatusFiat from './OrderStatusFiat';

describe('OrderStatusFiat', () => {
  it('renders correctly', () => {
    expect(shallow(<OrderStatusFiat status={0} />)).toMatchSnapshot();
    expect(shallow(<OrderStatusFiat status={8} />)).toMatchSnapshot();
    expect(shallow(<OrderStatusFiat status={11} />)).toMatchSnapshot();
    expect(shallow(<OrderStatusFiat status={12} />)).toMatchSnapshot();
    expect(shallow(<OrderStatusFiat status={13} />)).toMatchSnapshot();
    expect(shallow(<OrderStatusFiat status={14} />)).toMatchSnapshot();
    expect(shallow(<OrderStatusFiat status={15} />)).toMatchSnapshot();
    expect(shallow(<OrderStatusFiat status={16} />)).toMatchSnapshot();
  });
});
