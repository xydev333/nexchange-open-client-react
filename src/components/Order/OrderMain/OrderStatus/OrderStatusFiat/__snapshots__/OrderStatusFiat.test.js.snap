// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderStatusFiat renders correctly 1`] = `ShallowWrapper {}`;

exports[`OrderStatusFiat renders correctly 2`] = `ShallowWrapper {}`;

exports[`OrderStatusFiat renders correctly 3`] = `ShallowWrapper {}`;

exports[`OrderStatusFiat renders correctly 4`] = `ShallowWrapper {}`;

exports[`OrderStatusFiat renders correctly 5`] = `ShallowWrapper {}`;

exports[`OrderStatusFiat renders correctly 6`] = `ShallowWrapper {}`;

exports[`OrderStatusFiat renders correctly 7`] = `ShallowWrapper {}`;

exports[`OrderStatusFiat renders correctly 8`] = `ShallowWrapper {}`;
